compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -125,19 +141,34 @@ isFastPrimitive(CallExpr *call, bool isLocal) {
   case PRIM_AND_ASSIGN:
   case PRIM_OR_ASSIGN:
   case PRIM_XOR_ASSIGN:
-    if (!isCallExpr(call->get(2))) { // callExprs checked in calling function
-      if (!call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF) &&
-          !call->get(2)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF)) {
-        DEBUG_PRINTF(" *** OK (PRIM_MOVE 1): %s\n", call->primitive->name);
-        return true;
-      }
-    } else {
-      DEBUG_PRINTF(" *** OK (PRIM_MOVE 2): %s\n", call->primitive->name);
+    if (isCallExpr(call->get(2))) { // callExprs checked in calling function
       // Not necessarily true, but we return true because
       // the callExpr will be checked in the calling function
-      return true;
+      return FAST_AND_LOCAL;
+    } else {
+      bool arg1wide = call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF);
```

Comment:
Not something for this PR, but I wonder if it'd be possible to move optimizeOnClauses before insertWideReferences so we don't have to think about wide pointers here.